src/sentry/integrations/mixins/commit_context.py

```diff
@@ -1,17 +1,46 @@
 from __future__ import annotations
 
+from dataclasses import dataclass
+from datetime import datetime
 from typing import Any, Mapping, Protocol, Sequence
 
 from sentry.auth.exceptions import IdentityNotValid
 from sentry.models.identity import Identity
+from sentry.models.integrations.repository_project_path_config import RepositoryProjectPathConfig
 from sentry.models.repository import Repository
 from sentry.shared_integrations.exceptions import ApiError
 
 
+@dataclass
+class SourceLineInfo:
+    lineno: int
+    path: str
+    ref: str
+    repo: Repository
+    code_mapping: RepositoryProjectPathConfig
+
+
+@dataclass
+class CommitInfo:
+    commitId: str | None
+    committedDate: datetime
+    commitMessage: str | None
+    commitAuthorName: str | None
+    commitAuthorEmail: str | None
+
+
+@dataclass
+class FileBlameInfo(SourceLineInfo):
+    commit: CommitInfo
+
+
 class GetBlameForFile(Protocol):
     def get_blame_for_file(
         self, repo: Repository, filepath: str, ref: str, lineno: int
-    ) -> Sequence[Mapping[str, Any]]:
+    ) -> Sequence[Mapping[str, Any]] | None:
```

Comment:
```suggestion
    ) -> list[dict[str, object]] | None:
```
Return types are recommended to be explicit (e.g. `list`) unlike parameters of a function (e.g. `Sequence`)([link](https://typing.readthedocs.io/en/latest/source/best_practices.html#arguments-and-return-types)).